framework/include/restart/Restartable.h

```diff
@@ -177,11 +222,37 @@ Restartable::declareRestartableData(const std::string & data_name, Args &&... ar
   return declareRestartableDataWithContext<T>(data_name, nullptr, std::forward<Args>(args)...);
 }
 
+template <typename T, typename... Args>
+const T &
+Restartable::getRestartableData(const std::string & data_name, Args &&... args) const
+{
+  return getRestartableDataWithContext<T>(data_name, nullptr, std::forward<Args>(args)...);
+}
+
 template <typename T, typename... Args>
 T &
 Restartable::declareRestartableDataWithContext(const std::string & data_name,
                                                void * context,
                                                Args &&... args)
+{
+  return declareRestartableDataHelper<T>(data_name, context, std::forward<Args>(args)...).set();
+}
+
+template <typename T, typename... Args>
+const T &
+Restartable::getRestartableDataWithContext(const std::string & data_name,
+                                           void * context,
+                                           Args &&... args) const
+{
+  return std::as_const(
+      declareRestartableDataHelper<T>(data_name, context, std::forward<Args>(args)...).get());
```

Comment:
Look at https://stackoverflow.com/questions/123758/how-do-i-remove-code-duplication-between-similar-const-and-non-const-member-func

This is how this should be implemented. I think I led you poorly.

Reply:
Sounds good to me!